sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -166,6 +169,15 @@ public SentryId captureEnvelope(
     return sentryId;
   }
 
+  private void assignTraceContext(final @NotNull SentryEvent event) {
+    if (event.getThrowable() != null) {
+      final SpanContext spanContext = throwableToSpanContext.get(event.getThrowable());
+      if (spanContext != null && event.getContexts().getTrace() == null) {
```

Comment:
If these have public setters and are nullable we should navigate through considering they might be.

Reply:
Fixed.